src/components/OfferListTable.js

```diff
@@ -0,0 +1,97 @@
+import { useEffect, useState } from 'react';
+import { getitems, getitembyid, edititem, deleteitem } from '../APIs/ApiOffer';
+
+export default function OfferListTable({ isFilter, searchfilter, selectcategory }) {
+
+    const [offerList, setOfferList] = useState([]);
+
+    const loadData = () => {
+        getitems().then((response) => {
+            const data = response.data;
+            setOfferList(data);
+        }).catch(console.error());
+    }
+
+    useEffect(() => {
+        loadData();
+    }, []);
+
+    const toggleitems = (itemid) => {
+        getitembyid(itemid).then((response) => {
+            const toggle = response.data.isEnabled;
+            edititem(itemid, { isEnabled: !toggle }).then(() => { loadData() });
+        }).catch(console.error());
+    };
+
+    const deleteItems = (itemid) => {
+        deleteitem(itemid).then(() => { loadData() });
+    };
+
+    const filtercategory = () => {
+        switch (selectcategory) {
+            case "true": return offerList.filter(el => el.isEnabled === true)
+            case "false": return offerList.filter(el => el.isEnabled === false)
+            default: return offerList
+        };
+    };
+    const filterlist = () => {
+        if (isFilter === true) {
+            return filtercategory().filter(el => el.offer.toUpperCase().includes(searchfilter.toUpperCase()))
+        }
+        else {
+            return offerList
+        }
+    };
+
+    let totalImpressions = filterlist().filter(el => el.isEnabled === true).reduce((amt, cur) => { return amt + cur.impressions }, 0);
+    let totalConversions = filterlist().filter(el => el.isEnabled === true).reduce((amt, cur) => { return amt + cur.conversions }, 0);
+    let totalRevenue = filterlist().filter(el => el.isEnabled === true).reduce((amt, cur) => { return amt + cur.revenue }, 0).toFixed(2);
+    let totalConversionRate = filterlist().filter(el => el.isEnabled === true).reduce((amt, cur) => { return amt + ((cur.conversions / cur.impressions) * 100)}, 0).toFixed(2);
+
+    return (<>
```

Comment:
useEffect(() => {
         // do all the filters and calcuations
         
    }, [searchfilter, selectcategory]);

Reply:
Done